### PR TITLE
refactor/handlers

### PR DESCRIPTION
**Pull request by Caixetadev**
:warning: Work in progress

This pull request enhances the control flow of certain controllers and introduces a dedicated function for deleting documents and images.

### CODE REVIEW

**Comment by kevinanielsen** [comment]
Looks good so far - Let me know if you need help with anything!

**Comment by Caixetadev** [comment]
If you feel the pull request is becoming too extensive, please let me know, and I'll split it into smaller pull requests.

File: src/validations/validations.go
Comment (kevinanielsen) [inline src/validations/validations.go:1]: Could you filter the new filename using the filterFilename function? This is for security purposes.

**Review by kevinanielsen — CHANGES_REQUESTED** [review]

**Comment by kevinanielsen** [comment]
Also please let me know when it is no longer a WIP, so I can merge :)

File: src/validations/validations.go
Comment (Caixetadev) [reply src/validations/validations.go:1]: I'm using the Filename filter, I just chose not to put it inside this function. I just didn't put it inside the function, do you think it's better to call it there or leave it in the body of the handler? just below this validation

Comment (kevinanielsen) [reply src/validations/validations.go:1]: I honestly did not notice - The way you did it is perfect. Is this ready to merge or are you planning more changes?

**Review by kevinanielsen — APPROVED** [review]
Looks good!

File: src/validations/validations.go
Comment (Caixetadev) [reply src/validations/validations.go:1]: There are some handlers that can be improved, but if you want to merge these here now, it won't be so big

**Pull request merged**